src/hooks/chart.ts

```diff
@@ -1090,4 +1127,113 @@ const updateWithTraceColData = (
       }
     }
   }
+
 };
+
+/**
+ * 
+ * @param result data object to be updated
+ * @param trace from plot configs
+ * @param item from response object
+ * @param textArray array for hover text
+ * @param index provided for heatmap case for corresponding y axis values
+ * @param templateParams provided for heatmap to access url_parameters
+ * @returns 
+ */
+const updateHoverTemplateData = (
+  result: any,  
+  trace: Trace, 
+  item: any, 
+  textArray: any[], 
+  alertFunctions?: any,
+  index: number=0, 
+  templateParams?: PlotTemplateParams,
+  ) => {
+  const hovertemplate_display_pattern = trace.hovertemplate_display_pattern; // use trace info
+  if(hovertemplate_display_pattern){
+    const validLink=ConfigService.ERMrest._renderHandlebarsTemplate(hovertemplate_display_pattern, { $self: { data: item }, 
```

Comment:
change to `renderHandlebarsTemplate`. Functions beginning with `_` are intended to be private and only used in ermrestJS.